.azure-devops/azure-pipelines-pr.yml

```diff
@@ -0,0 +1,51 @@
+# runs on pull requests
+trigger: none
+
+# Environment variables
+variables:
+  - group: radius-angular
+  - name: chromaticProjectToken
+    value: $[variables.CHROMATIC_PROJECT_TOKEN]
+# triggers on pull requests against the following branches
+pr:
+  - main
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+  - checkout: self
+    persistCredentials: true
+    clean: true
+
+  - task: NodeTool@0
+    inputs:
+      versionSpec: "14.x"
+    displayName: "Install Node"
+
+  - script: npm ci --ignore-script
+    condition: ne(variables.CACHE_RESTORED, 'true')
+    displayName: "cache exists; Node clean install"
+
+  - task: Bash@3
+    displayName: "Lint and Test"
+    inputs:
+      targetType: "inline"
+      script: |
+        npm run lint
+        npm run test
+
+  - task: Bash@3
+    displayName: "Dry run of packaging"
+    inputs:
+      targetType: "inline"
+      script: |
+        npm run build:ds
+        cd dist/ds
+        npm pack --dry-run
```

Comment:
multiple commands in one script whaaaaat!? I'm glad to know this works

Reply:
Yes all you have to do is `script: | ` and write the commands one after another next line.

Sad thing is that the different jobs/stages I wrote before with just `script:` would never run and log the results. 

I am still unclear about whether the current working directory of the previous step is maintained after it completes, but separating out install, lint and build processes in separate jobs before proved that strategy took way too long and was checking itself out every time it went to the next job.

Reply:
Whether the CWD is preserved may depend on the task. Bash@ says:


> Working Directory	(Optional) Specify the working directory in which you want to run the command. If you leave it empty, the working directory is [$(Build.SourcesDirectory)](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops)

https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/bash?view=azure-devops